kii-core/kii_core.c

```diff
@@ -710,7 +715,11 @@ prv_http_request(
     if (body != NULL) {
         size_t body_len;
         char content_length[8];
-        body_len = kii_strlen(body);
+        if (binary_body_length != 0) {
+            body_len = binary_body_length;
+        } else {
```

Comment:
Using 0 to controlling statement is confusing.
Use minus or pass pointer.

Reply:
use negative value